changelog.d/15323.bugfix

```diff
@@ -0,0 +1 @@
+Fix joining rooms you have been unbanned from over federation. Introduced before Synapse bugs were tracked on Github. Contributed by Nico.
```

Comment:
```suggestion
Fix a long-standing bug preventing users from joining rooms, that they had been unbanned from, over federation. Contributed by Nico.
```

as much as 'before bugs were on github' is a fun timescale, I guess we should just put 'long-standing' like we usually do :p